### ISSUE TITLE
RoundCapper seems to be "working in reverse", shape is spiky not round

### ISSUE THREAD

**Issue by tv42** [golang/freetype#25]
![out](https://cloud.githubusercontent.com/assets/419909/12692479/59a2716a-c6ac-11e5-9743-628811443f67.png)

```go
package main

import (
	"image"
	"image/png"
	"log"
	"os"

	"github.com/golang/freetype/raster"
	"golang.org/x/image/math/fixed"
)

func main() {
	const (
		w = 400
		h = 400
	)
	var p raster.Path
	p.Start(fixed.P(150, 150))
	p.Add1(fixed.P(200, 250))
	p.Add1(fixed.P(300, 250))

	r := raster.NewRasterizer(w, h)
	r.UseNonZeroWinding = true
	r.AddStroke(p, fixed.I(80), raster.RoundCapper, raster.RoundJoiner)

	rgba := image.NewRGBA(image.Rect(0, 0, w, h))
	painter := raster.NewRGBAPainter(rgba)
	painter.SetColor(image.Black)
	r.Rasterize(painter)

	f, err := os.Create("out.png")
	if err != nil {
		log.Fatalf("cannot create file: %v", err)
	}
	defer f.Close()
	if err := png.Encode(f, rgba); err != nil {
		log.Fatalf("writing png: %v", err)
	}
	if err := f.Close(); err != nil {
		log.Fatalf("closing file: %v", err)
	}
}
```

**Comment by fogleman** [comment]
@tv42 Have you figured this out? I'm running into the same issue.

**Comment by tv42** [comment]
@fogleman No clue about the internals. I skimmed the repo just enough to think that the bug was probably introduced at the same time as the `math/fixed` change, but the commit in question was a bit too verbose to find anything relevant in.

**Comment by fogleman** [comment]
Any insight into this, @nigeltao ? I can try to fix it if you can point me in the right direction.

**Comment by nigeltao** [comment]
It's probably a bug introduced by https://github.com/golang/freetype/commit/856a70c3951a3ffe7a6c0dd5bb6729813eda12e3 or https://github.com/golang/freetype/commit/d91c1d6844f3b29492463cfb42cbf8d6e907d5ef

Sorry to not be more helpful. I've got a lot on my plate right now.

**Comment by fogleman** [comment]
I can confirm that the bug was introduced in 856a70c - worked properly before that.

**Comment by fogleman** [comment]
@tv42 @nigeltao I have submitted a pull request that fixes this. It's pretty simple: https://github.com/golang/freetype/pull/29

**Comment by nigeltao** [comment]
Fixed by PR #29, which has been merged.
